docs/plugin-api.asciidoc

```diff
@@ -0,0 +1,57 @@
+
+ifdef::env-github[]
+NOTE: For the best reading experience,
+please view this documentation at https://www.elastic.co/guide/en/apm/agent/java[elastic.co]
+endif::[]
+
+[[plugin-api]]
+== Plugin API
+The plugin API of the Elastic APM Java agent lets you add custom instrumentation to the agent, which
+the agent will automatically apply the same way as it applies the internally defined instrumentation.
+
+The plugin API is the OpenTelemetry API, plus a dependency to the Plugin SDK (apm-agent-plugin-sdk),
+and requires a version 1.31.0+ agent.
+
+[source,xml]
+.pom.xml
+----
+<dependency>
+    <groupId>co.elastic.apm</groupId>
+    <artifactId>apm-agent-plugin-sdk</artifactId>
+    <version>${elastic-apm.version}</version>
+</dependency>
+----
+
+[source,groovy]
+.build.gradle
+----
+compile "co.elastic.apm:apm-agent-plugin-sdk:$elasticApmVersion"
+----
+
+Replace the version placeholders with the
+link:https://search.maven.org/search?q=g:co.elastic.apm%20AND%20a:apm-agent-api[
+latest version from maven central]:
+image:https://img.shields.io/maven-central/v/co.elastic.apm/apm-agent-api.svg[Maven Central]
+
+An https://github.com/elastic/apm-agent-java-plugin-example[example repo] and an
+https://www.elastic.co/blog/create-your-own-instrumentation-with-the-java-agent-plugin[article]
+provide a detailed example of adding custom instrumentation for an application to the agent. An overview is
+
+1. subclass `co.elastic.apm.agent.sdk.ElasticApmInstrumentation`
+2. specify matchers the define which classes and methods will be instrumented
+3. add an instrumentation advice implementation
+4. create a META-INF/services/co.elastic.apm.agent.sdk.ElasticApmInstrumentation file which lists the
+fully qualified instrumentation class names, one class name per line
+5. create a plugin jar that includes the instrumentation classes, the `co.elastic.apm.agent.sdk.ElasticApmInstrumentation`
+file, and any dependencies (apart from the agent itself, though including the plugin API as shown above)
+6. start your application with the agent as normal, but additionally with the
+https://www.elastic.co/guide/en/apm/agent/java/current/config-core.html#config-plugins-dir[plugins_dir configuration option]
+set to a directory which includes the created plugin jar (and which should only hold plugin jars, as the agent will attempt
+to load any jar in that directory).
+
+[float]
+[[community]]
+=== Community Plugins
+
+To help our community, we've provided a <<community-plugins, page>> where you can list plugins you create that you think
```

Comment:
The docs build is failing because of an unresolved `community-plugins` reference. I think the issue is that you didn't include `docs/community-plugins.asciidoc` anywhere.